docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_source/_create_a_data_source.md

```diff
@@ -50,39 +50,44 @@ import DatasourceMethodReferenceTable from './_datasource_method_reference_table
 
    <DatasourceMethodReferenceTable/>
 
-   Once you have the method for your data's SQL dialect, you can call it with the previously defined Data Source name and connection string to create your Data Source.  The following example creates a PostgreSQL Data Source:
+   Once you have the method for your data's SQL dialect, you can call it with the previously defined Data Source name and connection details to create your Data Source.
+   
+   The following example creates a PostgreSQL Data Source using a connection string. This is the pattern most Data Sources follow:
 
    ```python title="Python" name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_source/postgres.py create data source"
    ```
-4. Optional. If you're connecting to Snowflake and want to use key-pair authentication instead of a password, pass the private key with `kwargs`. Note that a placeholder password is still required to pass the configuration validation, but the password will not be used if a `private_key` is provided.
+
+   Here is an example of creating a Snowflake Data Source using separate parameters for each connection detail and credential.
 
    ```python title="Python"
+   datasource_name = "my_new_snowflake_datasource"
+   account = "accountname.region"
+   user = "my_user"
+   database = "my_db"
+   schema = "my_schema"
+   warehouse = "my_wh"
+   role = "my_role"
    # For details on how to access your private key, refer to "Configure credentials" above 
-   connect_args = {"private_key": pkb} 
-
-   connection_details={
-      "account": "accountname.region",
-      "user": "my_user",
-      "role": "my_role",
-      "password": "placeholder_value", # must be provided to pass validation but will be ignored
-      "warehouse": "my_wh",
-      "database": "my_db",
-      "schema": "my_schema"
-   }
 
    data_source = context.data_sources.add_snowflake(
       name=datasource_name,
-      connection_string=connection_details,
-      kwargs={"connect_args": connect_args}
+      account=account,
+      user=user,
+      database=database,
+      schema=schema,
+      warehouse=warehouse,
+      role=role,
+      private_key=pkb
+   }
    )
    ```
    
    :::warning Private key serialized in File Data Context
-   If you're using a [File Data Context](/core/set_up_a_gx_environment/create_a_data_context.md), `kwargs` will be serialized to `great_expectations.yml`, including the private key.
+   If you're using a [File Data Context](/core/set_up_a_gx_environment/create_a_data_context.md), your Data Source configuration will be serialized to `great_expectations.yml`, including the private key.
```

Comment:
this was necessary since we didn't allow config substitution in arbitrary kwargs. I would suggest we just remove this part, unless we're including it for all sensitive fields on data sources. For instance, if you provide a password directly to a data source, it will be serialized out in your file data context as well.